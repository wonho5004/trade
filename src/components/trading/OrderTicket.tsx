'use client';

import { useEffect, useState } from 'react';
import type { ChangeEvent, FormEvent } from 'react';

import { useChartStore } from '@/stores/chartStore';

const defaultForm = {
  symbol: 'BTCUSDT',
  side: 'buy',
  quantity: '0.01',
  price: ''
};

export function OrderTicket() {
  const [form, setForm] = useState(defaultForm);
  const currentSymbol = useChartStore((state) => state.symbol);

  useEffect(() => {
    setForm((prev) => ({ ...prev, symbol: currentSymbol }));
  }, [currentSymbol]);

  const handleChange = (event: ChangeEvent<HTMLInputElement | HTMLSelectElement>) => {
    const { name, value } = event.target;
    setForm((prev) => ({ ...prev, [name]: value }));
  };

  const handleSubmit = (event: FormEvent<HTMLFormElement>) => {
    event.preventDefault();
    // 주문 제출 로직은 CCXT 주문 생성 모듈 연동 후 구현 예정입니다.
    console.info('주문 요청', form);
  };

  return (
    <form
      onSubmit={handleSubmit}
      className="flex w-full flex-col gap-4 rounded-lg border border-zinc-800 bg-zinc-950 p-6"
    >
      <div className="grid gap-2">
        <label className="text-sm text-zinc-400" htmlFor="symbol">
          거래쌍
        </label>
        <input
          id="symbol"
          name="symbol"
          value={form.symbol}
          onChange={handleChange}
          className="rounded border border-zinc-700 bg-zinc-900 px-3 py-2 text-sm text-zinc-100"
        />
      </div>
      <div className="grid gap-2">
        <label className="text-sm text-zinc-400" htmlFor="side">
          매매 방향
        </label>
        <select
          id="side"
          name="side"
          value={form.side}
          onChange={handleChange}
          className="rounded border border-zinc-700 bg-zinc-900 px-3 py-2 text-sm text-zinc-100"
        >
          <option value="buy">매수</option>
          <option value="sell">매도</option>
        </select>
      </div>
      <div className="grid gap-2">
        <label className="text-sm text-zinc-400" htmlFor="quantity">
          수량
        </label>
        <input
          id="quantity"
          name="quantity"
          value={form.quantity}
          onChange={handleChange}
          type="number"
          step="0.0001"
          min="0"
          className="rounded border border-zinc-700 bg-zinc-900 px-3 py-2 text-sm text-zinc-100"
        />
      </div>
      <div className="grid gap-2">
        <label className="text-sm text-zinc-400" htmlFor="price">
          지정가 (선택)
        </label>
        <input
          id="price"
          name="price"
          value={form.price}
          onChange={handleChange}
          type="number"
          step="0.1"
          min="0"
          className="rounded border border-zinc-700 bg-zinc-900 px-3 py-2 text-sm text-zinc-100"
        />
      </div>
      <button
        type="submit"
        className="rounded bg-emerald-500 px-4 py-2 text-sm font-semibold text-emerald-950 transition hover:bg-emerald-400"
      >
        주문 실행
      </button>
    </form>
  );
}
