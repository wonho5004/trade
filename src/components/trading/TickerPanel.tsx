'use client';

import { useEffect, useMemo } from 'react';
import { useShallow } from 'zustand/react/shallow';

import { useChartStore } from '@/stores/chartStore';
import { useMarketDataStore } from '@/stores/marketDataStore';

const formatNumber = (value: number | null, fractionDigits = 2) => {
  if (value === null || Number.isNaN(value)) {
    return '-';
  }
  return value.toLocaleString(undefined, {
    minimumFractionDigits: fractionDigits,
    maximumFractionDigits: fractionDigits
  });
};

export function TickerPanel() {
  const symbol = useChartStore((state) => state.symbol);
  const { status, lastPrice, priceChangePercent, highPrice, lowPrice, volume } = useMarketDataStore(
    useShallow((state) => ({
      status: state.status,
      lastPrice: state.lastPrice,
      priceChangePercent: state.priceChangePercent,
      highPrice: state.highPrice,
      lowPrice: state.lowPrice,
      volume: state.volume
    }))
  );
  const attachTicker = useMarketDataStore((state) => state.attach);

  useEffect(() => {
    if (!symbol) {
      return;
    }
    const detach = attachTicker(symbol);
    return detach;
  }, [symbol, attachTicker]);

  const isPositive = useMemo(() => (priceChangePercent ?? 0) >= 0, [priceChangePercent]);

  return (
    <section className="flex h-full flex-col justify-between rounded-lg border border-zinc-800 bg-zinc-950 p-6">
      <header className="flex items-baseline justify-between">
        <div>
          <p className="text-sm text-zinc-400">실시간 가격</p>
          <h2 className="text-lg font-semibold text-zinc-100">{symbol}</h2>
        </div>
        <span
          className={`text-sm font-semibold ${
            isPositive ? 'text-emerald-400' : 'text-rose-400'
          }`}
        >
          {status === 'connected' && priceChangePercent !== null
            ? `${formatNumber(priceChangePercent, 2)}%`
            : status === 'connecting'
            ? '연결 중...'
            : status === 'error'
            ? '에러'
            : '-'}
        </span>
      </header>

      <div className="mt-6 grid grid-cols-2 gap-4 text-sm text-zinc-400">
        <div>
          <p className="text-xs uppercase tracking-wide text-zinc-500">마지막 체결가</p>
          <p className="mt-2 text-base font-semibold text-zinc-100">
            {status === 'connected' ? formatNumber(lastPrice, 2) : '-'}
          </p>
        </div>
        <div>
          <p className="text-xs uppercase tracking-wide text-zinc-500">24h 고가</p>
          <p className="mt-2 text-base font-semibold text-zinc-100">{formatNumber(highPrice, 2)}</p>
        </div>
        <div>
          <p className="text-xs uppercase tracking-wide text-zinc-500">24h 저가</p>
          <p className="mt-2 text-base font-semibold text-zinc-100">{formatNumber(lowPrice, 2)}</p>
        </div>
        <div>
          <p className="text-xs uppercase tracking-wide text-zinc-500">24h 거래량</p>
          <p className="mt-2 text-base font-semibold text-zinc-100">{formatNumber(volume, 0)}</p>
        </div>
      </div>
    </section>
  );
}
