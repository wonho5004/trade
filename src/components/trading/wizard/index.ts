/**
 * Setup Wizard 컴포넌트 export
 */

export { SetupWizard } from './SetupWizard';
export { WelcomeStep } from './WelcomeStep';
export { ExperienceStep } from './ExperienceStep';
export { RecommendationStep } from './RecommendationStep';
export { CompleteStep } from './CompleteStep';
