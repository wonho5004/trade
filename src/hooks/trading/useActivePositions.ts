'use client';

import { useState, useEffect, useCallback } from 'react';
import type { ActivePositionSummary } from '@/types/trading/monitoring';

interface UseActivePositionsReturn {
  positions: ActivePositionSummary[];
  isLoading: boolean;
  error: string | null;
  closePosition: (id: string) => Promise<void>;
  refetch: () => Promise<void>;
}

export function useActivePositions(): UseActivePositionsReturn {
  const [positions, setPositions] = useState<ActivePositionSummary[]>([]);
  const [isLoading, setIsLoading] = useState(true);
  const [error, setError] = useState<string | null>(null);

  const fetchPositions = useCallback(async () => {
    setIsLoading(true);
    setError(null);

    try {
      const response = await fetch('/api/monitoring/positions');
      const data = await response.json();

      if (!response.ok) {
        // 401 에러는 조용히 처리 (로그인 필요)
        if (response.status === 401) {
          setError(null); // 에러 표시 안함
          setIsLoading(false);
          return;
        }
        throw new Error(data.error?.message || '포지션 조회에 실패했습니다');
      }

      setPositions(data.positions || []);
    } catch (err) {
      const message = err instanceof Error ? err.message : '알 수 없는 오류가 발생했습니다';
      setError(message);
      console.error('Failed to fetch positions:', err);
    } finally {
      setIsLoading(false);
    }
  }, []);

  const closePosition = useCallback(async (id: string) => {
    const response = await fetch(`/api/monitoring/positions/${id}/close`, {
      method: 'POST'
    });

    const data = await response.json();

    if (!response.ok) {
      throw new Error(data.error?.message || '포지션 청산에 실패했습니다');
    }

    // Refetch to update the list
    await fetchPositions();
  }, [fetchPositions]);

  // Initial fetch and periodic updates
  useEffect(() => {
    fetchPositions();

    // Poll every 5 seconds for real-time updates
    const interval = setInterval(fetchPositions, 5000);

    return () => clearInterval(interval);
  }, [fetchPositions]);

  return {
    positions,
    isLoading,
    error,
    closePosition,
    refetch: fetchPositions
  };
}
