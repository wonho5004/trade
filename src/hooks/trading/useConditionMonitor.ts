'use client';

import { useState, useEffect, useCallback } from 'react';
import type { ConditionEvaluation } from '@/types/trading/monitoring';

interface UseConditionMonitorReturn {
  evaluations: ConditionEvaluation[];
  isLoading: boolean;
  error: string | null;
  activeStrategies: number;
  refetch: () => Promise<void>;
}

export function useConditionMonitor(): UseConditionMonitorReturn {
  const [evaluations, setEvaluations] = useState<ConditionEvaluation[]>([]);
  const [activeStrategies, setActiveStrategies] = useState(0);
  const [isLoading, setIsLoading] = useState(true);
  const [error, setError] = useState<string | null>(null);

  const fetchEvaluations = useCallback(async () => {
    setIsLoading(true);
    setError(null);

    try {
      const response = await fetch('/api/monitoring/condition-evaluations?limit=20');
      const data = await response.json();

      if (!response.ok) {
        // 401 에러는 조용히 처리 (로그인 필요)
        if (response.status === 401) {
          setError(null); // 에러 표시 안함
          setIsLoading(false);
          return;
        }
        throw new Error(data.error?.message || '조건 평가 내역 조회에 실패했습니다');
      }

      setEvaluations(data.evaluations || []);
      setActiveStrategies(data.activeStrategies || 0);
    } catch (err) {
      const message = err instanceof Error ? err.message : '알 수 없는 오류가 발생했습니다';
      setError(message);
      console.error('Failed to fetch condition evaluations:', err);
    } finally {
      setIsLoading(false);
    }
  }, []);

  // Initial fetch and periodic updates
  useEffect(() => {
    fetchEvaluations();

    // Poll every 5 seconds for real-time condition monitoring
    const interval = setInterval(fetchEvaluations, 5000);

    return () => clearInterval(interval);
  }, [fetchEvaluations]);

  return {
    evaluations,
    isLoading,
    error,
    activeStrategies,
    refetch: fetchEvaluations
  };
}
