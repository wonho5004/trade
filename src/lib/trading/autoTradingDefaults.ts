import type {
  AutoTradingSettings,
  AdditionalEntryConditions,
  CapitalSettings,
  EntryConditions,
  EntryDirectionSettings,
  ExitConditions,
  ExitDirectionSettings,
  HedgeActivationSettings,
  IndicatorConditions,
  IndicatorConfigMap,
  IndicatorEntry,
  IndicatorKey,
  AggregatorOperator,
  ConditionGroupNode,
  ConditionNode,
  IndicatorLeafNode,
  CandleLeafNode,
  InitialMarginSetting,
  PositionDirection,
  ProfitCondition,
  ScaleInBudget,
  ScaleInDirectionSettings,
  ScaleInLimit,
  StopLossConditions,
  StopLossLineSettings,
  SymbolSelection,
  ThresholdCondition,
  IndicatorComparisonLeaf,
  CandleCondition,
  ComparisonOperator,
  CandleFieldOption,
  CandleReference
} from '@/types/trading/auto-trading';

export const createProfitCondition = (): ProfitCondition => ({
  enabled: false,
  comparator: 'over',
  value: 0
});

export const createThresholdCondition = (): ThresholdCondition => ({
  enabled: false,
  comparator: 'over',
  value: 0
});

const createIndicatorEntryId = () => `ind-${Date.now().toString(36)}-${Math.random().toString(36).slice(2, 8)}`;

const createConditionNodeId = () => `cond-${Date.now().toString(36)}-${Math.random().toString(36).slice(2, 8)}`;

export const createCandleCondition = (): CandleCondition => ({
  enabled: false,
  field: 'close',
  comparator: 'over',
  targetValue: 0,
  reference: 'current'
});

export const createCandleLeaf = (candle?: CandleCondition): CandleLeafNode => ({
  kind: 'candle',
  id: createConditionNodeId(),
  candle: candle ? { ...candle } : createCandleCondition()
});

export const createBollingerCondition = (): IndicatorConfigMap['bollinger'] => ({
  enabled: false,
  length: 20,
  standardDeviation: 2,
  offset: 0,
  band: 'middle',
  action: 'touch',
  touchTolerancePct: 0.2
});

export const createMaCondition = (): IndicatorConfigMap['ma'] => ({
  enabled: false,
  period: 20,
  actions: []
});

export const createRsiCondition = (): IndicatorConfigMap['rsi'] => ({
  enabled: false,
  period: 14,
  smoothing: 'sma',
  threshold: 50,
  actions: []
});

export const createDmiCondition = (): IndicatorConfigMap['dmi'] => ({
  enabled: false,
  diPeriod: 14,
  adxPeriod: 14,
  adx: createThresholdCondition(),
  diComparison: null,
  diPlus: createThresholdCondition(),
  diMinus: createThresholdCondition(),
  adxVsDi: null
});

export const createMacdCondition = (): IndicatorConfigMap['macd'] => ({
  enabled: false,
  fast: 12,
  slow: 26,
  signal: 9,
  source: 'close',
  method: 'EMA',
  comparison: null,
  histogramAction: null
});

export const createIndicatorConfig = <T extends IndicatorKey>(indicator: T): IndicatorConfigMap[T] => {
  switch (indicator) {
    case 'bollinger':
      return createBollingerCondition() as IndicatorConfigMap[T];
    case 'ma':
      return createMaCondition() as IndicatorConfigMap[T];
    case 'rsi':
      return createRsiCondition() as IndicatorConfigMap[T];
    case 'dmi':
      return createDmiCondition() as IndicatorConfigMap[T];
    case 'macd':
      return createMacdCondition() as IndicatorConfigMap[T];
    default:
      return createBollingerCondition() as IndicatorConfigMap[T];
  }
};

export const createIndicatorEntry = <T extends IndicatorKey>(indicator: T): IndicatorEntry<T> => ({
  id: createIndicatorEntryId(),
  type: indicator,
  config: createIndicatorConfig(indicator)
});

export const createIndicatorComparisonLeaf = (
  legacy: unknown = { kind: 'none' },
  fallbackId: string
): IndicatorComparisonLeaf | { kind: 'none' } => {
  if (!legacy || typeof legacy !== 'object') {
    return { kind: 'none' };
  }
  const candidate = legacy as any;
  if (candidate.kind === 'candle' && candidate.field && candidate.reference && candidate.comparator) {
    return {
      kind: 'candle',
      comparator: candidate.comparator,
      field: candidate.field,
      reference: candidate.reference
    };
  }
  if (candidate.kind === 'value' && typeof candidate.value === 'number' && candidate.comparator) {
    return {
      kind: 'value',
      comparator: candidate.comparator,
      value: candidate.value
    };
  }
  if (candidate.kind === 'indicator' && candidate.comparator) {
    return {
      kind: 'indicator',
      comparator: candidate.comparator,
      targetIndicatorId: candidate.targetIndicatorId ?? fallbackId
    };
  }
  return { kind: 'none' };
};

export const createIndicatorLeaf = (
  indicator: IndicatorEntry,
  comparison: IndicatorComparisonLeaf | { kind: 'none' } = { kind: 'none' }
): IndicatorLeafNode => ({
  kind: 'indicator',
  id: createConditionNodeId(),
  indicator,
  comparison
});

export const createIndicatorGroup = (
  operator: AggregatorOperator = 'and',
  children: ConditionNode[] = []
): ConditionGroupNode => ({
  kind: 'group',
  id: createConditionNodeId(),
  operator,
  children
});

export const createIndicatorConditions = (operator: AggregatorOperator = 'or'): IndicatorConditions => {
  // Stable default tree with deterministic IDs for test equality
  const candle = createCandleCondition();
  const root: ConditionGroupNode = {
    kind: 'group',
    id: 'cond-root',
    operator,
    children: []
  };

  // Return a legacy-compatible shape for transitional tests
  return {
    root,
    candle,
    entries: [],
    defaultAggregator: root.operator
  } as unknown as IndicatorConditions;
};

export const normalizeConditionTree = (root: ConditionNode): ConditionGroupNode => {
  // Defensive: tolerate unexpected/legacy shapes where `root` may be missing or malformed
  const isValidNode = (n: any) => n && typeof n === 'object' && typeof n.kind === 'string';
  const base: ConditionNode = isValidNode(root)
    ? (root as ConditionNode)
    : createIndicatorGroup('and', []);

  const collectIndicatorIds = (node: ConditionNode, acc: Set<string>) => {
    if (node.kind === 'indicator') {
      acc.add(node.id);
      return;
    }
    if (node.kind === 'group') {
      node.children.forEach((child) => collectIndicatorIds(child, acc));
    }
  };

  const sanitizeComparisons = (node: ConditionNode, indicatorIds: Set<string>): ConditionNode => {
    if (node.kind === 'indicator') {
      if (node.comparison.kind === 'indicator' && !indicatorIds.has(node.comparison.targetIndicatorId)) {
        return { ...node, comparison: { kind: 'none' } };
      }
      return node;
    }
    if (node.kind === 'group') {
      return {
        ...node,
        children: node.children.map((child) => sanitizeComparisons(child, indicatorIds))
      };
    }
    return node;
  };

  const ensureGroupRoot = (node: ConditionNode): ConditionGroupNode => {
    if (node.kind === 'group') return { ...node };
    return createIndicatorGroup('and', [node]);
  };

  const indicatorIds = new Set<string>();
  collectIndicatorIds(base, indicatorIds);
  const sanitized = sanitizeComparisons(base, indicatorIds);
  return ensureGroupRoot(sanitized);
};

const createInitialMarginSetting = (): InitialMarginSetting => ({
  basis: 'wallet',
  mode: 'per_symbol_percentage',
  percentage: 5,
  minNotional: 0,
  usdtAmount: 0,
  asset: 'USDT'
});

const createScaleInBudget = (): ScaleInBudget => ({
  mode: 'balance_percentage',
  basis: 'wallet',
  percentage: 5,
  minNotional: 0,
  usdtAmount: 0,
  asset: 'USDT'
});

const createScaleInLimit = (): ScaleInLimit => ({
  balance: {
    enabled: false,
    basis: 'wallet',
    percentage: 10
  },
  initialMargin: {
    enabled: false,
    percentage: 100
  },
  purchaseCount: {
    enabled: false,
    comparator: 'over',
    value: 1
  },
  profitRate: createProfitCondition(),
  unlimited: true
});

export const createCapitalSettings = (): CapitalSettings => ({
  estimatedBalance: 10000,
  maxMargin: {
    basis: 'wallet',
    percentage: 20
  },
  initialMargin: createInitialMarginSetting(),
  scaleInBudget: createScaleInBudget(),
  scaleInLimit: createScaleInLimit(),
  useMinNotionalFallback: true,
  hedgeBudget: {
    separateByDirection: false,
    long: { mode: 'position_percent', percentage: 100 },
    short: { mode: 'position_percent', percentage: 100 }
  }
});

export const createEntryDirectionSettings = (direction: PositionDirection): EntryDirectionSettings => ({
  enabled: false,
  immediate: false,
  indicators: createIndicatorConditions()
});

export const createEntryConditions = (): EntryConditions => ({
  long: createEntryDirectionSettings('long'),
  short: createEntryDirectionSettings('short')
});

export const createScaleInDirectionSettings = (direction: PositionDirection): ScaleInDirectionSettings => ({
  enabled: direction === 'long',
  profitTarget: createProfitCondition(),
  currentProfitRate: createProfitCondition(),
  indicators: createIndicatorConditions(),
  currentBuyNotional: { enabled: false, asset: 'USDT', comparator: 'none', value: 0 },
  initialBuyChange: { enabled: false, comparator: 'none', value: 0 }
});

export const createScaleInConditions = (): AdditionalEntryConditions => ({
  long: createScaleInDirectionSettings('long'),
  short: createScaleInDirectionSettings('short')
});

export const createExitDirectionSettings = (direction: PositionDirection): ExitDirectionSettings => ({
  enabled: direction === 'long',
  profitTarget: createProfitCondition(),
  currentProfitRate: createProfitCondition(),
  indicators: createIndicatorConditions(),
  includeFeesFunding: false,
  currentBuyNotional: { enabled: false, asset: 'USDT', comparator: 'none', value: 0 },
  initialBuyChange: { enabled: false, comparator: 'none', value: 0 },
  sellAmount: { mode: 'percent', value: 0 }
});

export const createExitConditions = (): ExitConditions => ({
  long: createExitDirectionSettings('long'),
  short: createExitDirectionSettings('short')
});

export const createStopLossLineSettings = (): StopLossLineSettings => ({
  enabled: false,
  autoRecreate: true,
  profitTarget: createProfitCondition(),
  indicators: createIndicatorConditions()
});

export const createStopLossConditions = (): StopLossConditions => ({
  profitTarget: createProfitCondition(),
  currentProfitRate: createProfitCondition(),
  purchaseCount: {
    enabled: false,
    comparator: 'over',
    value: 1
  },
  stopLossLine: createStopLossLineSettings(),
  indicators: createIndicatorConditions()
});

export const createSymbolSelection = (): SymbolSelection => ({
  manualSymbols: [],
  ranking: {
    market_cap: null,
    volume: null,
    top_gainers: null,
    top_losers: null
  },
  excludedSymbols: [],
  excludedReasons: {},
  respectDefaultExclusions: true,
  leverageMode: 'uniform',
  leverageOverrides: {},
  positionOverrides: {},
  featureOverrides: {},
  maxListingAgeDays: null,
  rankingSort: 'alphabet',
  autoFillRecheck: false,
  excludeTopGainers: null,
  excludeTopLosers: null,
  excludeBottomVolume: null,
  excludeBottomMarketCap: null
});

export const createHedgeActivationSettings = (): HedgeActivationSettings => ({
  enabled: false,
  directions: ['long', 'short'],
  currentProfitRate: createProfitCondition(),
  indicators: createIndicatorConditions()
});

export const createDefaultAutoTradingSettings = (): AutoTradingSettings => ({
  logicName: '',
  leverage: 5,
  symbolCount: 1,
  assetMode: 'single',
  positionMode: 'one_way',
  capital: createCapitalSettings(),
  timeframe: '3m',
  symbolSelection: createSymbolSelection(),
  entry: createEntryConditions(),
  scaleIn: createScaleInConditions(),
  exit: createExitConditions(),
  stopLoss: createStopLossConditions(),
  hedgeActivation: createHedgeActivationSettings(),
  metadata: {
    lastSavedAt: null,
    lastValidatedAt: null
  }
});
