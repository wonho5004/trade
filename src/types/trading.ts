export type OrderSide = 'buy' | 'sell';

export interface ExchangeCredentials {
  apiKey: string;
  secret: string;
  password?: string;
}

export interface OrderRequest {
  symbol: string;
  side: OrderSide;
  quantity: number;
  price?: number;
  type: 'market' | 'limit';
}
