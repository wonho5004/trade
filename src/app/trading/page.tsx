import { AppHeader } from '@/components/common/AppHeader';
import { OrderTicket } from '@/components/trading/OrderTicket';
import { TickerPanel } from '@/components/trading/TickerPanel';

export default function TradingPage() {
  return (
    <div className="flex min-h-screen flex-col gap-6 bg-zinc-950 text-zinc-100">
      <AppHeader />
      <main className="flex flex-1 flex-col gap-6 px-6 pb-10">
        <section className="grid gap-6 md:grid-cols-[3fr,2fr]">
          <TickerPanel />
          <OrderTicket />
        </section>
      </main>
    </div>
  );
}
