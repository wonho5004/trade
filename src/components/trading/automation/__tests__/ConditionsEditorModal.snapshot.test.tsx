// @ts-nocheck
import React from 'react';
import { render } from '@testing-library/react';

import { ConditionsEditorModal } from '../ConditionsEditorModal';
import { createIndicatorGroup } from '@/lib/trading/autoTradingDefaults';

describe('ConditionsEditorModal snapshots', () => {
  it('renders select mode (embedded) consistently', () => {
    const value: any = { root: createIndicatorGroup('or', []) };
    const { container } = render(
      <ConditionsEditorModal open title="조건 편집" value={value} onChange={() => {}} onClose={() => {}} initialMode="select" embedded />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders edit mode (embedded) consistently', () => {
    const value: any = { root: createIndicatorGroup('and', []) };
    const { container } = render(
      <ConditionsEditorModal open title="조건 편집" value={value} onChange={() => {}} onClose={() => {}} initialMode="edit" embedded />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

