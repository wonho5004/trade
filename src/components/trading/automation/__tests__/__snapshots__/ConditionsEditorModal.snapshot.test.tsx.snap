// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConditionsEditorModal snapshots renders edit mode (embedded) consistently 1`] = `
<div
  class="rounded-lg border border-zinc-800 bg-zinc-950"
>
  <div
    class="p-4 max-h-[75vh] overflow-auto"
  >
    <div
      class="space-y-4"
    >
      <div
        class="flex items-center justify-end"
      />
      <div
        class="space-y-4"
      >
        <div
          class="space-y-3"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`ConditionsEditorModal snapshots renders select mode (embedded) consistently 1`] = `
<div
  class="rounded-lg border border-zinc-800 bg-zinc-950"
>
  <div
    class="p-4 max-h-[75vh] overflow-auto"
  >
    <div
      class="space-y-4"
    >
      <p
        class="text-xs text-zinc-400"
      >
        지표 선택
      </p>
      <div
        class="grid grid-cols-2 gap-2 md:grid-cols-3"
      >
        <button
          aria-pressed="false"
          class="rounded px-3 py-2 text-left text-xs border border-zinc-700 text-zinc-200 hover:border-emerald-500/60 hover:text-emerald-200"
          type="button"
        >
          볼린저 밴드
        </button>
        <button
          aria-pressed="false"
          class="rounded px-3 py-2 text-left text-xs border border-zinc-700 text-zinc-200 hover:border-emerald-500/60 hover:text-emerald-200"
          type="button"
        >
          이동평균선
        </button>
        <button
          aria-pressed="false"
          class="rounded px-3 py-2 text-left text-xs border border-zinc-700 text-zinc-200 hover:border-emerald-500/60 hover:text-emerald-200"
          type="button"
        >
          RSI
        </button>
        <button
          aria-pressed="false"
          class="rounded px-3 py-2 text-left text-xs border border-zinc-700 text-zinc-200 hover:border-emerald-500/60 hover:text-emerald-200"
          type="button"
        >
          DMI / ADX
        </button>
        <button
          aria-pressed="false"
          class="rounded px-3 py-2 text-left text-xs border border-zinc-700 text-zinc-200 hover:border-emerald-500/60 hover:text-emerald-200"
          type="button"
        >
          MACD
        </button>
      </div>
      <div
        class="flex items-center gap-2"
      >
        <button
          class="rounded border border-emerald-500/60 px-3 py-1.5 text-xs font-semibold text-emerald-200 disabled:opacity-50"
          disabled=""
          type="button"
        >
          추가
        </button>
        <span
          class="text-[11px] text-zinc-500"
        >
          선택 후 ‘추가’를 눌러 아래 목록에 쌓입니다.
        </span>
      </div>
      <div
        class="h-px w-full bg-zinc-800/60"
      />
      <p
        class="text-xs text-zinc-400"
      >
        상태 조건
      </p>
      <div
        class="grid grid-cols-2 gap-2 md:grid-cols-4"
      >
        <button
          aria-pressed="false"
          class="rounded px-3 py-2 text-left text-xs border border-zinc-700 text-zinc-200 hover:border-sky-500/60 hover:text-sky-200"
          type="button"
        >
          현재 수익률(%)
        </button>
        <button
          aria-pressed="false"
          class="rounded px-3 py-2 text-left text-xs border border-zinc-700 text-zinc-200 hover:border-sky-500/60 hover:text-sky-200"
          type="button"
        >
          현재 마진
        </button>
        <button
          aria-pressed="false"
          class="rounded px-3 py-2 text-left text-xs border border-zinc-700 text-zinc-200 hover:border-sky-500/60 hover:text-sky-200"
          type="button"
        >
          매수 횟수
        </button>
        <button
          aria-pressed="false"
          class="rounded px-3 py-2 text-left text-xs border border-zinc-700 text-zinc-200 hover:border-sky-500/60 hover:text-sky-200"
          type="button"
        >
          진입 경과(일)
        </button>
      </div>
      <div
        class="flex items-center gap-2"
      >
        <button
          class="rounded border border-sky-500/60 px-3 py-1.5 text-xs font-semibold text-sky-200 disabled:opacity-50"
          disabled=""
          type="button"
        >
          상태 추가
        </button>
        <span
          class="text-[11px] text-zinc-500"
        >
          상태 조건은 새 그룹으로 추가됩니다.
        </span>
      </div>
    </div>
  </div>
</div>
`;
